manager/option.go

```diff
@@ -0,0 +1,79 @@
+package manager
+
+import (
+	"time"
+
+	"github.com/eapache/go-resiliency/retrier"
+)
+
+type Option interface {
```

Comment:
I think it would be simpler to have just `type Option func(m *Manager)` - most options could be simplified to inline the function. And there isn't really any other reasonable method into this interface 🤔

Reply:
Functional options pattern is implemented [here](https://github.com/strvcom/strv-backend-go-net/blob/master/http/response.go#L16) for example